runners/prism/java/src/main/java/org/apache/beam/runners/prism/PrismLocator.java

```diff
@@ -0,0 +1,253 @@
+      ZipInputStream zis = new ZipInputStream(from);
+      for (ZipEntry entry = zis.getNextEntry(); entry != null; entry = zis.getNextEntry()) {
+        InputStream in = ByteStreams.limit(zis, entry.getSize());
+        ByteStreams.copy(in, out);
+      }
+    } catch (IOException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  private static void copy(InputStream from, Path to) {
+    try {
+      ByteStreams.copy(from, Files.newOutputStream(to));
+    } catch (IOException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  private static void download(URL from, Path to) {
+    try {
+      ByteStreams.copy(from.openStream(), Files.newOutputStream(to));
+    } catch (IOException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  private static void validateShaSum512(URL shaSumReference, URL source) throws IOException {
+    try (InputStream in = shaSumReference.openStream()) {
+      String rawContent = new String(ByteStreams.toByteArray(in), StandardCharsets.UTF_8);
+      checkState(!Strings.isNullOrEmpty(rawContent));
+      String reference = "";
+      Iterator<String> split = Splitter.onPattern("\\s+").split(rawContent).iterator();
+      if (split.hasNext()) {
+        reference = split.next();
+      }
+      checkState(!Strings.isNullOrEmpty(reference));
+
+      HashCode toVerify = Hashing.sha512().hashBytes(ByteStreams.toByteArray(source.openStream()));
+      checkState(
+          reference.equals(toVerify.toString()),
+          "Expected sha512 derived from: %s does not equal expected: %s, got: %s",
+          source,
+          reference,
+          toVerify.toString());
+    }
+  }
+
+  private static String getNameWithoutExtension(String path) {
+    return org.apache.beam.vendor.guava.v32_1_2_jre.com.google.common.io.Files
+        .getNameWithoutExtension(path);
+  }
+
+  private String getSDKVersion() {
+    if (Strings.isNullOrEmpty(options.getPrismVersionOverride())) {
+      return RELEASE_INFO.getSdkVersion();
+    }
+    return options.getPrismVersionOverride();
+  }
+
+  private static String os() {
+    String result = mustGetPropertyAsLowerCase(OS_NAME_PROPERTY);
+    if (result.contains("mac")) {
+      return "darwin";
+    }
+    return result;
+  }
+
+  private static String arch() {
+    String result = mustGetPropertyAsLowerCase(ARCH_PROPERTY);
+    if (result.contains("aarch")) {
+      return "arm64";
+    }
+    return result;
+  }
+
+  private static String userHome() {
+    return mustGetPropertyAsLowerCase(USER_HOME_PROPERTY);
+  }
+
+  private static String mustGetPropertyAsLowerCase(String name) {
+    return checkStateNotNull(System.getProperty(name), "System property: " + name + " not set")
+        .toLowerCase();
+  }
+
+  private static void createDirectoryIfNeeded(Path path) throws IOException {
+    Path parent = path.getParent();
+    if (parent == null) {
+      return;
+    }
+    if (parent.toFile().exists()) {
+      return;
+    }
```

Comment:
I think all of this is handled by `Files.createDirectories`. The second check at least is for sure - https://docs.oracle.com/javase/8/docs/api/java/nio/file/Files.html#createDirectories-java.nio.file.Path-java.nio.file.attribute.FileAttribute...-